src/ArangoDBSnapshotStore.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * This file is part of the prooph/arangodb-snapshot-store.
+ * (c) 2016-2016 prooph software GmbH <contact@prooph.de>
+ * (c) 2016-2016 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\ArangoDB\SnapshotStore;
+
+use ArangoDBClient\Batch;
+use ArangoDBClient\Connection;
+use ArangoDBClient\Statement;
+use JsonSerializable;
+use Prooph\EventSourcing\Aggregate\AggregateType;
+use Prooph\EventSourcing\Snapshot\Snapshot;
+use Prooph\EventSourcing\Snapshot\SnapshotStore;
+
+final class ArangoDBSnapshotStore implements SnapshotStore
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * Custom sourceType to snapshot mapping
+     *
+     * @var array
+     */
+    private $snapshotCollectionMap;
+
+    /**
+     * @var string
+     */
+    private $defaultSnapshotCollectionName;
+
+    public function __construct(
+        Connection $connection,
+        array $snapshotCollectionMap = [],
+        string $defaultSnapshotCollectionName = 'snapshots'
+    ) {
+        $this->connection = $connection;
+        $this->snapshotCollectionMap = $snapshotCollectionMap;
+        $this->defaultSnapshotCollectionName = $defaultSnapshotCollectionName;
+    }
+
+    public function get(AggregateType $aggregateType, string $aggregateId): ?Snapshot
+    {
+        $collectionName = $this->getCollectionName($aggregateType);
+        $query = <<<EOF
+FOR s IN @@collection 
+    FILTER s._key == @aggregate_id && s.aggregate_type == @aggregate_type
+    SORT s.last_version DESC 
+    RETURN s
+EOF;
+
+        $statement = new Statement(
+            $this->connection, [
+                'query' => $query,
+                'bindVars' => [
+                    '@collection' => $collectionName,
+                    'aggregate_type' => $aggregateType->toString(),
+                    'aggregate_id' => $aggregateId,
+                ],
+            ]
+        );
+
+        $cursor = $statement->execute();
+
+        if (! $cursor->getCount()) {
+            return null;
+        }
+
+        $result = $cursor->current();
+
+        return new Snapshot(
+            $aggregateType,
+            $aggregateId,
+            $result->is_json
+                ? '' // TODO create aggregate from array
+                : unserialize($result->aggregate_root, ['allowed_classes' => [$aggregateType->toString()]]),
```

Comment:
@prolic If we use `JsonSerializable` we need a mechanism to create the aggregate root from an `array`. Any suggestions? Should we use an own interface where the user have to implement `fromArray` too or do I miss something? I've added an extra field in `save` to mark if it was saved as `json`.

We should also add the `allowed_classes` option to `unserialize` for security reasons (PHP Object Injection Vulnerability). Should we add other classes like `\Prooph\EventSourcing\AggregateRoot` too? 

/cc @codeliner